model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/entities/SessionKey.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.models.sessions.infinispan.entities;
+
+import java.util.Objects;
+
+import org.infinispan.protostream.annotations.Proto;
+import org.infinispan.protostream.annotations.ProtoTypeId;
+import org.keycloak.marshalling.Marshalling;
+import org.keycloak.models.utils.KeycloakModelUtils;
+
+@ProtoTypeId(Marshalling.SESSION_KEY)
+@Proto
+public record SessionKey(String uuid, boolean offline) {
```

Comment:
Does it make sense to just store this as a `UUID` given that it's wire format is just two longs?

Reply:
To reduce the network traffic and storage space, yes. 

My concerns are around queries and indexes. It is unclear if I need to index this field or if `UUID` can be indexed. Also, the Keycloak code uses `String` everywhere. Example: 
https://github.com/keycloak/keycloak/blob/18a6c790113998bf039f46d6c71ffb668e961036/server-spi/src/main/java/org/keycloak/models/UserSessionModel.java#L38

Reply:
If we want indexing, we probably would have to add our own UUIDAdapter. I'm happy leaving this as a String.

Reply:
ProtoStream ships the UUIDAdapter and I'm not sure which one the Infinispan Server will pick. I would rather stay with String for now :+1: